common/src/Exceptions.h

```diff
@@ -165,6 +165,22 @@ namespace TrenchBroom {
         FileFormatException(const String& str) throw() : ExceptionStream(str) {}
         ~FileFormatException() throw() {}
     };
+
+    class ConditionFailedException : public ExceptionStream<ConditionFailedException> {
+    public:
+        ConditionFailedException() throw() {}
+        ConditionFailedException(const String& str) throw() : ExceptionStream(str) {}
+        ConditionFailedException(const char *file, const int line, const char *condition, const String& message) throw() : ExceptionStream() {
+            *this << file << ":" << line << ": Condition '" << condition << "' failed: " << message;
+        }
+        ~ConditionFailedException() throw() {}
+    };
 }
 
+// These are ugly but necessary to stringify an expression, see: https://en.wikipedia.org/wiki/C_preprocessor#Token_stringification
```

Comment:
Macros.h can't include Exceptions.h because there would be a circular dependency; Exceptions.h includes StringUtils.h which includes Macros.h.

Not sure what the best workaround is.. it looks like Exceptions.h is just using StringUtils to get the `String` and `StringStream` typedefs, it would probably be easy to make Exceptions.h use std::string and std::stringstream directly.